### ISSUE TITLE
Hardware related problems and a different approach

### ISSUE THREAD

**Issue by Peersen** [denjello/pi-DJex#9]
Safe shutdown:
Just to make this clear, the original version was stable for weeks on my desk.
But in the real world application the Pi will be hard reseted several times a day.
It would be wise to attach a battery and make the Pi shutdown safe once Power is turned off.

Noise:
The used cheap USB condenser microphone`s audio signal had enormous ammount of noise in it.
SNR was only about 40db, using a seperate USB hub helped a little.
A USB cable with an induktor could help some more.
It could be worth trying a dynamic microphone instead.

Reverse engineer Data loging instead of Audio input:
A complete different approach would be to use a real SPL meter with an USB or serial connection.
At least the affordable no name products would need reverse engineering to be done.
Providing a Python module that interfaces with those meters could be even more usefull to the public or future projects.

Someone already did that:
http://opensource.ebswift.com/RaspiMonitor/wensn/
http://sigrok.org/wiki/Tondaj_SL-814

these meters could be identical to the above mentioned ones:
http://www.ebay.de/itm/USB-Larm-Schallpegel-Messgerat-Schallpegelmesser-Dezibel-Schallpegel-130dB-Neu-/251688614616?pt=LH_DefaultDomain_77&hash=item3a99cf7ad8

http://www.amazon.de/Familymall-30-130dB-Schallpegelmessger-Schallpegelmesser-Schallpegel/dp/B00TH1SJMY/ref=sr_1_8?ie=UTF8&qid=1425404540&sr=8-8&keywords=schallpegelmessger%C3%A4t+USB

http://www.amazon.de/SODIAL-Schallpegel-Messgeraet-Pegelmesser-Schallpegelmessgeraet/dp/B008YONMDG/ref=pd_sim_sbs_diy_6?ie=UTF8&refRID=1KP7RK395QA34B1RV7F6

http://www.amazon.de/gp/offer-listing/B008YONMDG/ref=dp_olp_new?ie=UTF8&condition=new

**Comment by dasantonym** [comment]
the app will be much more tolerant to hard resets and power outages. there will be almost no more logging (writing to the sd card), so corruption is less likely (given that all unnecessary other services will be removed).

@denjello had an idea for safe shutdowns i think.

+1000 for the external sound monitoring through hardware!

wouldn't an analog one enable us to feed the level meter's input to gpios/arduino?
http://www.amazon.de/Teufel-Schallpegel-Messger%C3%A4t/dp/B000ZLS53S/ref=pd_sim_sbs_236_5?ie=UTF8&refRID=0FD1WPCHVGJJEJRNR9YT

**Comment by denjello** [comment]
yeah, the idea about safe shutdowns was actually an entirely RO system.

Here is a pretty thorough solution (on arch)
https://gist.github.com/yeokm1/8b0ffc03e622ce011010

**Comment by denjello** [comment]
can't we pipe the sound output from the mixer to the PI's external sound card and then output from the sound card to the PA? this way we could automatically limit or play this song: https://www.youtube.com/watch?v=BROWqjuTM0g

**Comment by dasantonym** [comment]
+10000 for the rickroll mode (could be optional and i want the trololo mode)! if we solve the latency problem first...

**Comment by denjello** [comment]
Here is an audio card (needs kernel patch on raspbian apparently) with line-in / line-out and onboard microphones.

http://www.linuxuser.co.uk/reviews/wolfson-audio-card-for-raspberry-pi-review

here is apparently a newer version:

http://www.element14.com/community/docs/DOC-71261/l/cirrus-logic-audio-card-for-raspberry-pi-a-plus-and-b-plus#overview

**Comment by dasantonym** [comment]
but was the audio card a problem in the first place? i thought it was more the microphone and the dedicated noise measurement tools just work better out of the box.
